addons/web/static/src/js/views/list/list_renderer.js

```diff
@@ -874,7 +874,7 @@ var ListRenderer = BasicRenderer.extend({
         if (node.attrs.readOnly) {
             $th.addClass('oe_read_only');
         }
-        if (!field) {
+        if (!field || node.attrs.nolabel) {
```

Comment:
in form renderer, we do `node.attrs.nolabel !== '1'` (which is already a bit naive), I suggest to keep the same semantics.